server/text_generation_server/layers/fp8.py

```diff
@@ -200,56 +272,89 @@ def get_linear(self, bias: torch.Tensor):
         # memory. Can be non-contiguous when we e.g. expand from scalars.
         self.weight_scale = self.weight_scale.contiguous()
         return get_fp8_linear().from_fp8(
-            self.weight, self.weight_scale, self.activation_scale_ub, bias, self.dtype
+            self.weight,
+            self.weight_scale,
+            self.input_scale,
+            self.activation_scale_ub,
+            bias,
+            self.dtype,
         )
 
 
 class Fp8Linear(torch.nn.Module):
```

Comment:
Would it be cleaner to have a separate Fp8LinearRocm?

Reply:
Maybe, it depends a bit on how much conditional code we end up with. We did separate FP8 Marlin for this reason.